src/dqe_idx.erl

```diff
@@ -227,6 +228,28 @@ metrics(Collection) ->
 
 %%--------------------------------------------------------------------
 %% @doc
+%% Returns a list of metric path suffixes of `Depth' that are prefixed
+%% by the given probe `Prefix', which can also be empty.
+%%
+%% For example:
+%% metrics(<<"collection">>, <<>>, 1) -> [<<"base">>].
+%% metrics(<<"collection">>, <<"base">>, 1) -> [<<"cpu">>].
+%% metrics(<<"collection">>, <<>>, 2) -> [[<<"base">>.<<"cpu">>]].
+%%
+%% Actual results are binary-encoded in `dproto' encoding.
+%% @end
+%%--------------------------------------------------------------------
+-spec metrics(Collection::collection(),
+              Prefix::metric(),
+              Depth::pos_integer()) ->
+                    {ok, [Metric::metric()]} |
```

Comment:
so in

## DQE

* dqe all inputs to dqe_idx seems to be **lists** format (this goes hand in hand with non lookup metrics they are represented in list format too)
* The result in dqe is used in the **binary** format (and converted as required) as ddb_connection requires binary.
* The naming functions require **lists** (and convert as required form binary)


## FE

* metrics list requires **lists** and converts them form binary
* namespace listing **seems** to use **binary** as it uses a base64 encoded string as part of the URL.
* tag listing **seems** to use **binary** as it uses a base64 encoded string as part of the URL.
* value listing **seems** to use **binary** as it uses a base64 encoded string as part of the URL.

## Proxy

* indexer has **lists** and converts them to binary to fit
* prometheus protocl has **lists** and converts them to binary to fit


All in all only the get and the http api seem to use binary, given that the data comes in a list format it feels better to use lists everywhere and convert as needed instead of using binary and convert the lists. What's your thought?